fuse/data/visualizer/visualizer_base.py

```diff
@@ -25,17 +34,56 @@ def _show(self, vis_data: List):
         :param vis_data: preprocessed visualization items to display
         """
         raise "should implement abstract method"
+    
+    @abstractmethod
```

Comment:
Why abstract?

Reply:
it's part of Daniel's design, I think he wanted common API to all visualizers